homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -446,6 +522,48 @@ def native_value(self) -> int | None:
         return self.entity_description.value_fn(system_consumption)
 
 
+class EnvoyProductionPhaseEntity(EnvoySystemSensorEntity):
+    """Envoy phase production entity."""
+
+    entity_description: EnvoyProductionSensorEntityDescription
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state of the sensor."""
+        if (
+            self.entity_description.on_phase is None
+            or self.data.system_production_phases is None
+        ):
+            return None
```

Comment:
Is this check needed here? I think we avoid creating the entity (or should) if its going to return here

Reply:
No check is not needed indeed, only for the Linters not liking the None part and avoiding assert.

Reply:
used
```py
        if TYPE_CHECKING:
            assert self.entity_description.on_phase
            assert self.data.system_production_phases
```